// react-native-compressor/lib/commonjs/utils/Uploader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.backgroundUpload = exports.UploaderHttpMethod = exports.UploadType = void 0;
var _reactNative = require("react-native");
var _Main = require("../Main");
var _index = require("./index");
const CompressEventEmitter = new _reactNative.NativeEventEmitter(_Main.Compressor);
let UploadType = /*#__PURE__*/function (UploadType) {
  UploadType[UploadType["BINARY_CONTENT"] = 0] = "BINARY_CONTENT";
  UploadType[UploadType["MULTIPART"] = 1] = "MULTIPART";
  return UploadType;
}({});
exports.UploadType = UploadType;
let UploaderHttpMethod = /*#__PURE__*/function (UploaderHttpMethod) {
  UploaderHttpMethod["POST"] = "POST";
  UploaderHttpMethod["PUT"] = "PUT";
  UploaderHttpMethod["PATCH"] = "PATCH";
  return UploaderHttpMethod;
}({});
exports.UploaderHttpMethod = UploaderHttpMethod;
const backgroundUpload = async (url, fileUrl, options, onProgress) => {
  const uuid = (0, _index.uuidv4)();
  let subscription;
  try {
    if (onProgress) {
      subscription = CompressEventEmitter.addListener('uploadProgress', event => {
        if (event.uuid === uuid) {
          onProgress(event.data.written, event.data.total);
        }
      });
    }
    if (_reactNative.Platform.OS === 'android' && fileUrl.includes('file://')) {
      fileUrl = fileUrl.replace('file://', '');
    }
    const result = await _Main.Compressor.upload(fileUrl, {
      uuid,
      method: options.httpMethod,
      headers: options.headers,
      uploadType: options.uploadType,
      ...options,
      url
    });
    return result;
  } finally {
    // @ts-ignore
    if (subscription) {
      subscription.remove();
    }
  }
};
exports.backgroundUpload = backgroundUpload;
